app/src/main/java/com/github/factotum_sdp/factotum/ui/roadbook/RoadBookViewAdapter.kt

```diff
@@ -30,30 +34,82 @@ class RoadBookViewAdapter(
     }
     private val asyncList = AsyncListDiffer(this, differCallback)
 
+    // Inflate a new view hierarchy according to fragment_destrecord.xml design
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecordViewHolder {
         val view =
             LayoutInflater.from(parent.context)
                 .inflate(R.layout.fragment_destrecord, parent, false)
         return RecordViewHolder(view)
     }
+
+    /**
+     * Updates the async list displayed by this RoadBookViewAdapter
+     */
     fun submitList(ls: List<DestinationRecord>) {
         asyncList.submitList(ls)
     }
 
+    // Bind each displayed Record to the corresponding current async list slot
     override fun onBindViewHolder(holder: RecordViewHolder, position: Int) {
         val item = asyncList.currentList[position]
-        holder.idView.text = item.rate.toString()
-        holder.contentView.text = item.destName
+        val context = holder.itemView.context
+
+        // Format and bind the TextViews
+        holder.destID.text = item.destID
+        holder.timeStamp.text =
+            timestampStringFormat(item.timeStamp, context.getString(R.string.arrival_text_view))
+        holder.waitingTime.text =
+            waitingTimeStringFormat(item.waitingTime, context.getString(R.string.wait_text_view))
+        holder.rate.text =
+            rateStringFormat(item.rate, context.getString(R.string.rate_text_view))
+        holder.actions.text =
+            actionsStringFormat(item.actions, context.getString(R.string.actions_text_view))
+    }
+
+    private fun rateStringFormat(rate: Int, label: String): String {
+        return "$label : $rate"
+    }
+
+    // Only displayed if there is a non zero waiting time
+    private fun waitingTimeStringFormat(waitTime: Int, label: String): String {
+        if (waitTime != 0)
+            return "$label : $waitTime"
+        return ""
+    }
+
```

Comment:
Could benefit from comments describing the exact format you are building. Ex : dd/mm/yyyy for a date